### ISSUE TITLE
[Question] Can we validate a v1 token using this library?

### ISSUE THREAD

**Issue by mohsinhere** [AzureAD/microsoft-identity-web#444]
I have a v1 token with "iss": "https://sts.windows.net/{tenantid}/"

Microsoft.Identity.Web worked fine when I was using a v2 token. 

But when I use a v1 token I'm getting a "The signature key was not found" error when I try to hit the api.

My Azure AD Config is below:

```
"AzureAppRegistration": {
    "Instance": "https://login.microsoftonline.com/",
    "TenantId": "<my tenantid>",
    "ClientId": "<my client id for app registration>",
    "Domain": "[mydomain].onmicrosoft.com"
  }
```

Authentication code is:

`services.AddProtectedWebApi(Configuration, "AzureAppRegistration");`

Can anyone advise if I need to update my config to validate a v1 token?

Thanks!

**Comment by jmprieur** [comment]
@mohsinhere do you mean:
- a v1.0 token (acquired for a Web API that is using the AAD v2.0 endpoint? 
- Or a token acquired for your Web API that would be using the AAD v1.0 endpoint. ?

**If the earlier**, that's a bug, and we want to know (do you have a repro? this works for me. 

**If the later**, this is by design as Microsoft.Identity.Web is to support the v2.0 endpoint,
See https://github.com/AzureAD/microsoft-identity-web/blob/41195539af65836b464b063c186842896ab15e0e/src/Microsoft.Identity.Web/WebApiExtensions/MicrosoftIdentityWebApiAuthenticationBuilderExtensions.cs#L173

However, I'd think you could try to override it by writing something like this:

```CSharp
services.AddMicrosoftWebApiAuthentication(Configuration);
services.Configure<JwtBearerOptions>(JwtBearerDefaults.AuthenticationScheme, options =>
{
 options.Authority = options.Authority.Replace("/v2.0", string.Empty);
});

```

but we have not tried.

**Comment by mohsinhere** [comment]
@jmprieur 

The app registration for which I'm requesting for a token is configured to issue v1 token (the config value in the manifest).

All I need is, I need to use this token to call my API which uses Microsoft.Identity.Web.

PS: The reason I'm not changing the token version to 2 is that this same app registration is used to authenticate to an Application Proxy. So if I switch to v2, authenticating to the App Proxy would fail. 

The code sample that you posted doesn't seem to work... not sure if I should use an older library. Can you shed some light on this if that's the case?

Thanks!

**Comment by pmaytak** [comment]
Hi @mohsinhere, have you resolved this question?

Just to confirm - if you have 
1. a web API that uses Microsoft Identity Web 
2. set `"accessTokenAcceptedVersion": 1,` in manifest in AAD app registration for this web API

The API will accept that v1 access token (which has `"iss": "https://sts.windows.net/{tenantid}/"` as you mentioned)

**Comment by jmprieur** [comment]
@mohsinhere 
If you look at this sample: https://github.com/Azure-Samples/active-directory-dotnet-native-aspnetcore-v2/tree/master/1.%20Desktop%20app%20calls%20Web%20API and run the app creation scripts but don't do the manual steps, you'll see that the API requests a v1.0 access tokens and this works fine.

Closing this issue as we think we've answered.
Feel free to reopen if you disagree.

**Comment by jennyf19** [comment]
Included in [0.4.0-preview release](https://github.com/AzureAD/microsoft-identity-web/releases/tag/0.4.0-preview)
